### ISSUE TITLE
Build fails for iPhone simulator

### ISSUE THREAD

**Issue by pravincar** [framinosona/Xamarin.Nordic.DFU.iOS#2]
Hi @framinosona 

Thanks for putting these bindings together and sharing your work.

The nuget package version 4.7.1.50 works great on iPhone but the project fails to build for iPhone simulator.

`error MT5209 : Native linking error : warning: ld: warning: ignoring file xxx/obj/iPhoneSimulator/Debug/device-builds/iphone 8-13.5/mtouch-cache/iOSDFULibrary.framework/iOSDFULibrary, missing required architecture x86_64 in file xxx/obj/iPhoneSimulator/Debug/device-builds/iphone 8-13.5/mtouch-cache/iOSDFULibrary.framework/iOSDFULibrary (2 slices)ignoring file xxx/fdaapp.iOS/obj/iPhoneSimulator/Debug/device-builds/iphone 8-13.5/mtouch-cache/ZIPFoundation.framework/ZIPFoundation, missing required architecture x86_64 in file xxx/obj/iPhoneSimulator/Debug/device-builds/iphone 8-13.5/mtouch-cache/ZIPFoundation.framework/ZIPFoundation (2 slices)` 

I do not have much experience with Objective C bindings for Xamarin however let me know if this is something you need help with and I can dig into the problem further.

**Comment by alundgren93** [comment]
I'm also getting a lot of these errors. Any known workaround?

**Comment by alundgren93** [comment]
I created the binding library myself. Adding the [Protocol] attribute solves this issue. I'll try and create a PR once I get a chance.

**Comment by framinosona** [comment]
Hey @alundgren93, thanks for that. But Actually I didn't really fix it since I don't understand why you would want to have Nordic DFU Library on a Simulator build ... But if adding the [Protocol] attribute solves it and doesn't break the iPhone build then yes send a PR :)

**Comment by pravincar** [comment]
Thanks @alundgren93 and @framinosona .
I eventually took the approach to conditionally skip the service that has the DFU dependency from the simulator build.
However, I think it would be useful to get the simulator build to work, if that is straightforward.

**Comment by SuperCorks** [comment]
I agree with this issue. Apps do more than one thing and it's a bit of a nuisance to define conditions in the code to skip one service on simulator build (I'm doing the same thing). @framinosona, I don't know if you've already started to work on this, but I can give it a shot if you haven't. Let me know.
